iati/version.py

```diff
@@ -189,3 +197,297 @@ def next_decimal(self):
+
+    """
+    def wrapper(*args, **kwargs):
+        """Act as a wrapper to ensure a version number is a Decimal that exists.
+
+        Raises:
+            ValueError: If the input version is not a known Decimal iati.Version.
+
+        """
+        version = args[0]
+
+        if not _is_known(version):
+            raise ValueError('{0} is not a known version of the IATI Standard in a standardised representation.'.format(repr(version)))
+
+        return input_func(*args, **kwargs)
+
+    return wrapper
+
+
+def allow_possible_version(input_func):
+    """Decorate function by ensuring values specified to represent a Version can actually do so.
+
+    In terms of value:
+    * Permitted values representing an Integer or Decimal Version in a known format will remain unchanged.
+    * None will remain unchanged, as a way of representing all versions.
+    * strings, integers and Decimals with values that cannot represent a Version will cause a ValueError.
+    * Values of types other than string, Decimal, integer and iati.Version will cause a TypeError.
+
+    Args:
+        input_func (function): The function to decorate. Takes the `version` argument as its first argument.
+
+    Returns:
+        function: The input function, wrapped such that the return value is known to represent some IATI Version Number.
+
+    """
+    def wrapper(*args, **kwargs):
+        """Act as a wrapper to ensure a value represents a possible version number.
+
+        Raises:
+            TypeError: If the input version is not an iati.Version, string, Decimal or integer.
+            ValueError: If the input version is a string, Decimal or Integer, but the value cannot represent a Version Number.
+
+        """
+        version = args[0]
+
+        _prevent_non_version_representations(version)
+
+        return input_func(*args, **kwargs)
+
+    return wrapper
+
+
+def decimalise_integer(input_func):
+    """Decorate function by converting input version numbers to a standardised format Decimal Version.
+
+    In terms of value:
+    * Decimal Versions will remain unchanged.
+    * Integer Versions will return the latest Decimal Version within the Integer.
+
+    In terms of type:
+    * strings and Decimals will become iati.Versions.
+    * iati.Versions will remain unchanged.
+
+    Args:
+        input_func (function): The function to decorate. Takes the `version` argument as its first argument.
+
+    Returns:
+        function: The input function, wrapped such that it is called with a iati.Version representing a Decimal Version.
+
+    """
+    def wrapper(*args, **kwargs):
+        """Act as a wrapper to convert input Integer Version numbers to a standardised format Decimal Version."""
+        version = _decimalise_integer(args[0])
+
+        return input_func(version, *args[1:], **kwargs)
+
+    return wrapper
+
+
+def standardise_decimals(input_func):
```

Comment:
Agree that this (and a couple of the other decorators) don't have wonderful names.
Using the word `decimal` doesn't help due to this also being a datatype that can be used to represent some possible Minor Versions.

`decimal_to_version` doesn't quite work since it sounds like it converts an arbitrary `decimal.Decimal` to an `iati.Version`. Unhelpfully, I've not got any much better suggestions, though.

Reply:
How about simply `to_version` or `convert_to_version`??  Just thinking that the word `standardise` implies that this function is does something more directly related to the Standard?

Reply:
Those options are a bit too general and miss the information that it doesn't do anything to major version numbers.

`normalise_decimals` maybe? It removes any reference to `standard` in the name, is kinda similar to `standardise`, and context should make it clear enough that this isn't about database normalisation.
(`normalise`: to bring or return to a normal or standard condition or state)

Reply:
a62dab878363c17457b2aa5f14e4e5fdcaa78942